lib/spack/spack/cmd/buildcache.py

```diff
@@ -609,7 +609,7 @@ def get_concrete_spec(args):
 
     if spec_str:
         try:
-            spec = Spec(spec_str)
```

Comment:
If this isn't working with hashes there's a more serious regression issue somewhere in the codebase, I'm worried this is a bandaid on a missing limb.

Reply:
Yeah, I don't know enough about this codebase to say what the most correct fix would be. However, the goal with this fix was to make it consistent with the "spack buildcache create" command, which uses the same function (find_matching_specs) and does work with hashes now. Open to suggestion as I am new to this code and common patterns.

Reply:
@becker33 is there anything further you'd like done with this CR? Or a specific concern you have? I understand the bandaid issue, but this is actually making this code consistent with the rest of this file.